c10/cuda/CUDACachingAllocator.h

```diff
@@ -125,9 +128,39 @@ struct SegmentInfo {
   int64_t active_size = 0;
   cudaStream_t stream = 0;
   bool is_large = false;
+  MempoolId_t owner_private_pool_id = {0, 0};
   std::vector<BlockInfo> blocks;
 };
 
+struct BlockState {
+  int device;
+  cudaStream_t stream;
```

Comment:
who owns the cudaStream_t

Reply:
Will look into - this is the same ownership structure as Block... not sure exactly

Reply:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/cuda/CUDAGraph.h#L62